GoogleUtilities/FileDataStorage/GULStorageHeartbeat.m

```diff
@@ -35,14 +35,46 @@ - (instancetype)initWithFileURL:(NSURL *)fileURL {
   return self;
 }
 
+- (nullable NSMutableDictionary *)getDictionary {
+  NSString *jsonString = [NSString stringWithContentsOfURL:self.fileURL
+                                                  encoding:NSUTF8StringEncoding
+                                                     error:nil];
+  NSError *jsonError;
+  if (jsonString == nil) {
+    return [NSMutableDictionary dictionary];
+  }
+  NSData *objectData = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
+  NSMutableDictionary *json = [NSJSONSerialization JSONObjectWithData:objectData
+                                                              options:NSJSONReadingMutableContainers
+                                                                error:&jsonError];
+  if (!json) {
+    NSLog(@"Got an error: %@", jsonError);
+  }
+  return json;
+}
+
+- (BOOL)writeDictionary:(NSMutableDictionary *)dictionary error:(NSError **)outError {
```

Comment:
I can see the following issues with the current storage API:
1. All its clients should be aware of the dictionary structure (what keys and values types and meaning are)
2. One client with a bug can break entire storage. 
3. Any dictionary related logic update should be taken into account by all clients

I would propose the following API to hide more implementation details form clients:
`- (nullable NSDate *)heartbeatDateForTag:(NSString *)tag`
`- (BOOL)setHearbeatDate:(NSDate *)date forTag:(NSString *)tag`
In this case we can change the way we store data without affecting the class clients